File: Arrays/Done/MissingNumber.c
//http://www.geeksforgeeks.org/find-the-missing-number/
//Given a set of numbers betweeen integers A and B, exactly one of these integers is missing. (A<B)
//There are no duplicates in the list. Find the missing number
//
//Time Complexity  = 2*O(n)
//Space Complexity = O(1)
//Streaming Algorithm

#include<stdio.h>

int main()
{
    int arr[100];
    int A; int B;
    scanf("%d %d", &A, &B);

    int N=B-A;
    int i, j, k;
    int xor=0;
    for(i=A;i<=B;i++)
        xor=xor^i;

    int xorans=0;
    for(i=A;i<B;i++){
        scanf("%d", &N);
        xorans^=N;
    }
    printf("%d is the missing number.. .", xorans^xor);


    return 0;
}

File: TheGrind/2Week2/14MissingNumber1toN/MissingNumber.java
//http://www.geeksforgeeks.org/find-the-missing-number/
//
//Given a list of N-1 numbers from 1 to N find the missing number.
//Input: Array of N-1 numbers
//Output: Missing Number
import java.util.Scanner;

public class MissingNumber{
    public static void main(String[] args){
        Scanner in = new Scanner(System.in);

        int T = in.nextInt();
        int N;
        int temp, result, i, j;

        for(i=0;i<T;i++){
            N=in.nextInt();
            result=0;

            for(j=0;j<N-1;j++){
                temp=in.nextInt();
                result=result^temp;
                result=result^(j+1);
            }
            result=result^N;
            System.out.println(result);
            
        }
    }
}
